app/web/src/newhotness/DeleteModal.vue

```diff
@@ -0,0 +1,129 @@
+            <div class="text-xs my-xs">
+              <span class="flex flex-col">
+                <strong>Delete</strong>
+              </span>
+              <span class="flex flex-col">
+                If this component already exists on HEAD, it will be marked for
+                deletion and removed when this change set is applied. Otherwise,
+                the component will be removed deleted immediately.
+              </span>
+            </div>
+          </VormInputOption>
+          <VormInputOption :value="DeleteMode.Remove">
+            <div class="text-xs my-xs">
+              <span class="flex flex-col">
+                <strong>Remove</strong>
+              </span>
+              <span class="flex flex-col">
+                The component will be removed from this view. It will remain in
+                other views. However, if this is the only view that the
+                component exists in, the component will not be removed from the
+                view.
+              </span>
+            </div>
+          </VormInputOption>
+        </VormInput>
+      </div>
+
+      <div class="flex gap-sm">
+        <VButton label="Cancel" tone="shade" variant="ghost" @click="close" />
+        <VButton
+          class="flex-grow"
+          icon="trash"
+          label="Confirm"
+          tone="destructive"
+          @click="onConfirm"
+        />
+      </div>
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts" setup>
+import * as _ from "lodash-es";
+import {
+  Modal,
+  useModal,
+  VButton,
+  VormInput,
+  VormInputOption,
+  Icon,
+} from "@si/vue-lib/design-system";
+import { onBeforeUnmount, onMounted, ref } from "vue";
+import { BifrostComponentInList } from "@/workers/types/entity_kind_types";
+import { getAssetIcon, getAssetColor } from "./util";
+
+const components = ref<BifrostComponentInList[]>([]);
+
+const borderStyle = (component: BifrostComponentInList) => {
+  const color = getAssetColor(component.schemaCategory);
+  return `border-color: ${color}`;
+};
+
+const modalRef = ref<InstanceType<typeof Modal>>();
+const { open: openModal, close: closeModal } = useModal(modalRef);
+
+function open(selectedComponents: BifrostComponentInList[]) {
+  components.value = selectedComponents;
+  openModal();
+}
+function close() {
+  components.value = [];
+  closeModal();
+}
+
+const mode = ref(DeleteMode.Delete);
+
+async function onConfirm() {
+  close();
+  emit("delete", mode.value);
+}
+
+onMounted(() => {
+  window.addEventListener("keydown", onKeyDown);
+});
+onBeforeUnmount(() => {
+  window.removeEventListener("keydown", onKeyDown);
+});
+
+const onKeyDown = async (e: KeyboardEvent) => {
+  if (e.key === "Enter" && modalRef.value?.isOpen) {
+    onConfirm();
+  }
+};
+
+const emit = defineEmits<{
+  (e: "delete", value: DeleteMode): void;
+}>();
+
+defineExpose({ open, close });
+</script>
+
+<script lang="ts">
+export enum DeleteMode {
+  Delete,
+  Remove,
+}
+</script>
```

Comment:
Need a separate script to expose this an a separate `.ts` file did not make sense.